internal/daemon/daemon.go

```diff
@@ -695,13 +730,27 @@ func (d *Daemon) RebootIsMissing(st *state.State) error {
 	return state.ErrExpectedReboot
 }
 
+// finalize is used by v1Finalize to allow the Daemon to stop when
+// launched with Finalizer set to true.
+func (d *Daemon) finalize() {
+	if d.finalizer == nil {
+		return
+	}
+	select {
```

Comment:
Why is the select...default needed when `d.finalizer` is a 1-buffered channel? Why not `close()` an unbuffered channel instead of trying to send?

Reply:
can't close a closed channel, this means that the /v1/finalize api can be called as many times as it wants and only works once without needing more complextity